File: Decorator/Decorator.Exemplo/FP/CalculosSalario.cs
namespace Decorator.Exemplo.FP
{
    public static class CalculosSalario
    {
        public static double CalcularSalarioMensal(double valorPorHora)
            => valorPorHora * 40 * 5;

        public static double DescontarImpostos(double salario)
            => salario - (salario * 0.15);

        public static double DescontarPlanoSaude(double salario)
            => salario - 600;
    }

    
}

File: Decorator/Decorator.Exemplo.FSharp/CalculosSalario.fs
module CalculosSalario

let calcularSalarioMensal valorPorHora =
    valorPorHora * 40.0 * 5.0

let descontarImpostos salario =
    salario - (salario * 0.15)

let descontarPlanoSaude salario =
    salario - 600.0

